src/pkg/utils/helpers/slice.go

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package helpers provides generic helper functions with no external imports
+package helpers
+
+// EqualFunc defines a type for a function that determines equality between two elements of type T.
+type EqualFunc[T any] func(a, b T) bool
+
+// MergeSlices merges two slices, s1 and s2, and returns a new slice containing all elements from s1
+// and only those elements from s2 that do not exist in s1 based on the provided equal function.
+func MergeSlices[T any](s1, s2 []T, equals EqualFunc[T]) []T {
```

Comment:
If we are going to create a new slice.go file all of the slice functions from misc should move here too.